collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -396,17 +405,20 @@ public void recordBean(
 
       Receiver receiver = new Receiver();
       JmxScraper scraper = new JmxScraper(config.jmxUrl, config.username, config.password, config.ssl, config.whitelistObjectNames, config.blacklistObjectNames, receiver);
+      List<MetricFamilySamples> mfsList = new ArrayList<MetricFamilySamples>();
       long start = System.nanoTime();
       double error = 0;
-      try {
-        scraper.doScrape();
-      } catch (Exception e) {
-        error = 1;
-        StringWriter sw = new StringWriter();
-        e.printStackTrace(new PrintWriter(sw));
-        LOGGER.severe("JMX scrape failed: " + sw.toString());
+      if ((config.startDelaySeconds == 0) ||
```

Comment:
Accepted, I did note that in my comment. It now throws an IllegalStateException and registers a scrape error